### PR TITLE
Test Ruby 3.0 /3.1

### PR DESCRIPTION
**Pull request by poorndm**
Signed-off-by: poornima <poorndm@progress.com>

 mixi library for subprocess management, output collection

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: .expeditor/verify.pipeline.yml [hunk 1]
```diff
@@ -11,45 +11,45 @@ expeditor:
 
 steps:
 
-- label: run-lint-and-specs-ruby-2.4
+- label: run-lint-and-specs-ruby-2.5
```
Comment (tas50) [inline .expeditor/verify.pipeline.yml:14]: Make sure to also drop ruby 2.4 in the gemspec

File: .expeditor/verify.pipeline.yml [hunk 2]
```diff
@@ -11,45 +11,45 @@ expeditor:
...
+- label: run-lint-and-specs-ruby-3.1
   command:
     - .expeditor/run_linux_tests.sh rake
   expeditor:
     executor:
       docker:
-        image: ruby:3.0-buster
+        image: ruby:3.1
 
 - label: run-specs-windows
```
Comment (PrajaktaPurohit) [inline .expeditor/verify.pipeline.yml:54]: Do we know the version of Ruby for the windows box?

Comment (PrajaktaPurohit) [reply .expeditor/verify.pipeline.yml:54]: Looks like we are not testing this on Ruby 3.1 : https://buildkite.com/chef-oss/chef-mixlib-shellout-main-verify/builds/230#3c137b79-6a73-4f77-aad6-ab518b16303b/97-680 causing https://github.com/chef/ohai/pull/1738 to fail on windows.

File: .expeditor/verify.pipeline.yml (hunk 1 above)
Comment (PrajaktaPurohit) [reply .expeditor/verify.pipeline.yml:14]: Are we ready to drop ruby 2.4? Or do we need to pin in in places if it is beings used elsewhere?

Comment (PrajaktaPurohit) [reply .expeditor/verify.pipeline.yml:14]: We are ready to drop 2.4

**Review by PrajaktaPurohit — APPROVED** [review]
Looks like we are good to go after a rebase!

**Comment by johnmccrae** [comment]
Sorry, I polluted your branch. However, I copied out all the working goodness to a new branch and merged all the changes. It's all working now.

**Comment by johnmccrae** [comment]
translation - you can close this PR.

**Pull request closed without merging**